extra/3-magic-8-ball.js

```diff
@@ -59,8 +66,26 @@ function shakeBall() {
   This function should expect to be called with any value which was returned by the shakeBall function.
 */
 function checkAnswer(answer) {
-  //Write your code in here
+  let answerIteration;
+  for(let i = 0; i < answersArray.length; i++) {
+    if(answer === answersArray[i]) {
+      answerIteration = i;
+    }
+  }
+  //refactored my code below to make it into a ternary operator
+  return answerIteration < 5 ? "very positive" : answerIteration >=5 && answerIteration < 10 ? "positive"
+  : answerIteration >=10 && answerIteration < 15 ? "negative" : "very negative";
 }
+/**if(answerIteration < 5) {
+  return "very positive";
+} else if(answerIteration >= 5 && answerIteration < 10) {
+  return "positive";
+} else if(answerIteration >=10 && answerIteration < 15) {
+  return "negative"
+} else {
+  return "very negative";
+}**/
+
```

Comment:
**Reducing complexity**

- This is actually more readable, you can try use a bunch of `if` statements
- Oh avoid multiple `return` for `if` or `if...else` statements, try to assign the values, so that code flow can be read more linearly - it's more a special JavaScript style

Hint:
```js
let message;

if(answerIteration < 5 ) {
  message = "very positive";
} 

if(answerIteration >= 5 && answerIteration < 10 ) {
  message = "positive";
} 
// etc ...

return message;
```

Reply:
That's something I've learned since, I was writing a lot of extra code that isn't actually needed. It's great to come back to this so many weeks later and see the difference between this, and how I write code now.